isis/src/control/objs/BundleUtilities/BundleObservation.cpp

```diff
@@ -783,6 +783,761 @@ namespace Isis {
+    if (nPointingCoefficients == 0) {
+      nPointingCoefficients = 1;
+      useDefaultPointing = true;
+    }
+
+    // Force number of position and pointing parameters to each be 3 (X,Y,Z; RA,DEC,TWI)
+    // so we can always output a value for them
+    int nPositionParameters = 3 * nPositionCoefficients;
+    int nPointingParameters = 3 * nPointingCoefficients;
+    int nParameters = nPositionParameters + nPointingParameters;
+
+    coefX.resize(nPositionCoefficients);
+    coefY.resize(nPositionCoefficients);
+    coefZ.resize(nPositionCoefficients);
+    coefRA.resize(nPointingCoefficients);
+    coefDEC.resize(nPointingCoefficients);
+    coefTWI.resize(nPointingCoefficients);
+
+    if (m_instrumentPosition) {
+      if (!useDefaultPosition) {
+        m_instrumentPosition->GetPolynomial(coefX, coefY, coefZ);
+      }
+      // Use the position's center coordinate if not solving for spacecraft position
+      else {
+        const std::vector<double> centerCoord = m_instrumentPosition->GetCenterCoordinate();
+        coefX[0] = centerCoord[0];
+        coefY[0] = centerCoord[1];
+        coefZ[0] = centerCoord[2];
+      }
+    }
+
+    if (m_instrumentRotation) {
+      if (!useDefaultPointing) {
+        m_instrumentRotation->GetPolynomial(coefRA, coefDEC, coefTWI);
+
+      }
+      // Use the pointing's center angles if not solving for pointing (rotation)
+      else {
+        const std::vector<double> centerAngles = m_instrumentRotation->GetCenterAngles();
+        coefRA[0] = centerAngles[0];
+        coefDEC[0] = centerAngles[1];
+        coefTWI[0] = centerAngles[2];
+
+      }
+    }
+
+    // for convenience, create vectors of parameters names and values in the correct sequence
+    std::vector<vector<double> > finalParameterValues;
+    QStringList parameterNamesList;
+    QStringList correctionUnitList;
+
+    QString str("%1(%2)  ");
+
+    finalParameterValues.push_back(coefX);
+    finalParameterValues.push_back(coefY);
+    finalParameterValues.push_back(coefZ);
+    for (int i = 0; i < imagePosParams.size(); i++) {
+          if (nPositionCoefficients > 0) {
+          for (int j = 0; j < nPositionCoefficients;j++) {
+            if (j == 0) {
+              parameterNamesList.append(str.arg(imagePosParams[i]).arg("km"));
+              correctionUnitList.append("m");
+            } //end inner-if
+
+            else {
+              switch(j) {
+              case 1:
+                parameterNamesList.append( str.arg("     ").arg("km/s") );
+                correctionUnitList.append("m/s");
+                break;
+              case 2:
+                parameterNamesList.append( str.arg("     ").arg("km/s^2") );
+                correctionUnitList.append("m/s^2");
+                break;
+              case 3:
+                parameterNamesList.append( str.arg("     ").arg("km/s^3") );
+                correctionUnitList.append("m/s^3");
+                break;
```

Comment:
Remove case 2 and case 3: they are not necessary.

Reply:
Fixed.